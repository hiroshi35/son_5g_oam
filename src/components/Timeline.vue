<template>
  <div class="timeline_wrapper">
    <ul class="timeline">
      <slot></slot>
    </ul>
  </div>
</template>

<style>
  .timeline {
    list-style: none;
    padding: 20px 0 20px;
    position: relative;
  }

  .timeline:before {
    top: 0;
    bottom: 0;
    position: absolute;
    content: " ";
    width: 1px;
    background-color: #c5c5c5;
    left: 50%;
    margin-left: -1.5px;
  }

  .timeline > li {
    margin-bottom: 20px;
    position: relative;
  }

  .timeline > li:before,
  .timeline > li:after {
    content: " ";
    display: table;
  }

  .timeline > li:after {
    clear: both;
  }

  .timeline > li:before,
  .timeline > li:after {
    content: " ";
    display: table;
  }

  .timeline > li:after {
    clear: both;
  }

  .timeline > li > .timeline-panel {
    width: 46%;
    float: left;
    border: 1px solid #d4d4d4;
    border-radius: 0px;
    padding: 20px;
    position: relative;
    background: #fff;
    -webkit-box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12);
    box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12);
  }

  .timeline > li > .timeline-panel:before {
    position: absolute;
    top: 26px;
    right: -15px;
    display: inline-block;
    border-top: 15px solid transparent;
    border-left: 15px solid #ccc;
    border-right: 0 solid #ccc;
    border-bottom: 15px solid transparent;
    content: " ";
  }

  .timeline > li > .timeline-panel:after {
    position: absolute;
    top: 27px;
    right: -14px;
    display: inline-block;
    border-top: 14px solid transparent;
    border-left: 14px solid #fff;
    border-right: 0 solid #fff;
    border-bottom: 14px solid transparent;
    content: " ";
  }

  .timeline > li > .timeline-badge {
    border: 2px solid #eee;

    color: #fff;
    width: 50px;
    height: 50px;
    line-height: 47px;
    font-size: 1.4em;
    text-align: center;
    position: absolute;
    top: 16px;
    left: 50%;
    margin-left: -25px;
    background-color: #999999;
    z-index: 100;
    border-top-right-radius: 50%;
    border-top-left-radius: 50%;
    border-bottom-right-radius: 50%;
    border-bottom-left-radius: 50%;
  }

  .timeline > li > .timeline-badge > .date {
    font-size: 15px;
    font-family: Lato, sans-serif;
    font-weight: 300;
    color: #7d868c;
    text-align: left;
    position: absolute;
    top: 0;
    right: -70px;

  }
  @media only screen and (min-width: 472px) {
    .timeline > li.timeline-inverted > .timeline-badge > .date {
      text-align: right;
      color: #7d868c;
      position: absolute;
      top: 0;
      left: -110px;
      right: 50px;

    }

    .timeline > li.timeline-inverted > .timeline-panel {
      float: right;
    }

    .timeline > li.timeline-inverted > .timeline-panel:before {
      border-left-width: 0;
      border-right-width: 15px;
      left: -15px;
      right: auto;
    }

    .timeline > li.timeline-inverted > .timeline-panel:after {
      border-left-width: 0;
      border-right-width: 14px;
      left: -14px;
      right: auto;
    }
  }
  @media only screen and (max-width: 472px) {
    .timeline:before {
      top: 0;
      bottom: 0;
      position: absolute;
      content: " ";
      width: 1px;
      background-color: #c5c5c5;
      left: 70%;
      margin-left: -1.5px;
    }
    .timeline > li > .timeline-badge {
      left: 70%;
    }
    .timeline .timeline-panel{
      width: 60% !important;
    }
  }

  .timeline-badge.primary {
    background-color: #2e6da4 !important;
  }

  .timeline-badge.success {
    background-color: #3f903f !important;
  }

  .timeline-badge.warning {
    background-color: #f0ad4e !important;
  }

  .timeline-badge.danger {
    background-color: #d9534f !important;
  }

  .timeline-badge.info {
    background-color: #5bc0de !important;
  }

  .timeline-badge.blue {
    background-color: #65a4da !important;
  }

  .timeline-badge.green {
    background-color: #79d679 !important;
  }

  .timeline-badge.yellow {
    background-color: #f7bd6c !important;
  }

  .timeline-badge.red {
    background-color: #dc5e5a !important;
  }

  .timeline-badge.teal {
    background-color: #3ab39b !important;
  }

  .timeline-badge.purple {
    background-color: #976ad6 !important;
  }

  .timeline-title {
    margin-top: 0;
    color: inherit;
    font-family: Lato, sans-serif;
    font-weight: 300;
  }

  .timeline-body > p,
  .timeline-body > ul {
    margin-bottom: 0;
  }

  .timeline-body > p + p {
    margin-top: 5px;
  }
</style>
