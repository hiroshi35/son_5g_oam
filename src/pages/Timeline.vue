<template>
  <div>
    <div class="container-fluid">
      <page-header title="Timeline"></page-header>
      <div class="row">
        <div class="col-md-12">
          <timeline>
            <timeline-item title="Have a meeting with Will" date="1 hour ago" badge-color="green"
                           badge-icon='<i class="fa fa-calendar-check-o" aria-hidden="true"></i>'>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquam aspernatur consequuntur eveniet id in laboriosam nisi quaerat quas quia repellendus. Assumenda incidunt labore nesciunt nihil, nostrum quis repellendus. Fugit, voluptas.
              </p>
              <button class="btn btn-outlined-teal right-angular margin-top-10">Add Event</button>

            </timeline-item>
            <timeline-item title="Break" date="2 hours ago" :inverted="true">
              <p>
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquid cupiditate eveniet magni maxime! Consectetur cupiditate dolorem, facere ipsum nobis officiis quaerat quos. Aspernatur et nam natus nisi voluptas. Nulla, repudiandae.
              </p>

            </timeline-item>
            <timeline-item title="Send Code to Mike" date="Yesterday" badge-color="red"
                           badge-icon='<i class="fa fa-code" aria-hidden="true"></i>'>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Accusamus blanditiis dolorem esse hic ratione? Aspernatur atque doloremque enim, ex ipsum iste magnam nihil possimus quas qui rem sequi veritatis voluptatum!
              </p>
            </timeline-item>
            <timeline-item title="Recharge Batteries" date="30.04.2016" :inverted="true"
                           badge-icon='<i class="fa fa-battery-quarter" aria-hidden="true"></i>' badge-class="purple">
              Lorem ipsum dolor sit amet, consectetur adipisicing elit. Harum laudantium magni possimus repellat voluptas! Dolore et in ipsa libero nihil rem sapiente sint tempora voluptatum. Aliquam aspernatur facere officia repellat!
            </timeline-item>
            <timeline-item title="Deadline" date="14.04.2016" badge-color="yellow">
              <p>
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquid cupiditate eveniet magni maxime! Consectetur cupiditate dolorem, facere ipsum nobis officiis quaerat quos. Aspernatur et nam natus nisi voluptas. Nulla, repudiandae.
              </p>
              <div class="row">
                <div class="col-md-12">
                  <button class="btn btn-outlined-yellow right-angular margin-top-10 btn-block">Tasks</button>
                </div>
              </div>

            </timeline-item>
          </timeline>
        </div>
      </div>
    </div>
    <!-- /.container-fluid -->

  </div>
</template>

<script>
  import Timeline from '../components/Timeline.vue'
  import TimelineItem from '../components/TimelineItem.vue'

  export default {
    components: {
      Timeline,
      TimelineItem
    }
  }
</script>
