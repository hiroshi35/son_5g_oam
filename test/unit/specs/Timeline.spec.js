import { shallow } from 'vue-test-utils'
import Timeline from '@/components/Timeline.vue'

describe('Timeline.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(Timeline)
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should render correct class', () => {
    expect(wrapper.classes()).toContain('timeline_wrapper')
  })
})
