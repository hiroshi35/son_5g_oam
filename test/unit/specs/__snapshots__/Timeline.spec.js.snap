// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline.vue has the expected html structure 1`] = `
<div
  class="timeline_wrapper"
>
  <ul
    class="timeline"
  />
</div>
`;
